kernel/memory/src/paging/mapper.rs

```diff
@@ -867,3 +867,104 @@ pub fn mapped_pages_unmap(
 pub fn mapper_from_current() -> Mapper {
     Mapper::from_current()
 }
+
+
+/// An immutably borrowed [`MappedPages`] object that derefs to `&T`.
+///
+/// When dropped, the borrow ends and the contained `MappedPages` is dropped and unmapped.
+pub struct BorrowedMappedPages<T: FromBytes> {
+    ptr: NonNull<T>,
+    mp: MappedPages,
+}
+impl<T: FromBytes> Deref for BorrowedMappedPages<T> {
+    type Target = T;
+    fn deref(&self) -> &T {
+        // SAFETY:
+        // ✅ The pointer is properly aligned because we use `NonNull`.
```

Comment:
`NonNull` doesn't guarantee that the pointer is aligned. `NonNull::dangling()` does, but we don't use the dangling pointer.

The guarantee of the pointer being properly aligned should come from the fact that `as_type` returns a valid reference to `T` but, as I mention below, that may not be entirely true.

Reply:
Fixed now